src/tilesource.js

```diff
@@ -709,6 +709,166 @@ $.TileSource.prototype = {
+     */
+    downloadTileStart: function (context) {
+        context.image = new Image();
+
+        context.image.onload = function(){
+            context.finish(true);
+        };
+        context.image.onabort = context.image.onerror = function() {
+            context.finish(false, "Image load aborted");
+        };
+
+        // Load the tile with an AJAX request if the loadWithAjax option is
+        // set. Otherwise load the image by setting the source proprety of the image object.
+        if (context.loadWithAjax) {
+            context.request = $.makeAjaxRequest({
+                url: context.src,
+                withCredentials: context.ajaxWithCredentials,
+                headers: context.ajaxHeaders,
+                responseType: "arraybuffer",
+                postData: context.postData,
+                success: function(request) {
+                    var blb;
+                    // Make the raw data into a blob.
+                    // BlobBuilder fallback adapted from
+                    // http://stackoverflow.com/questions/15293694/blob-constructor-browser-compatibility
+                    try {
+                        blb = new window.Blob([request.response]);
+                    } catch (e) {
+                        var BlobBuilder = (
+                            window.BlobBuilder ||
+                            window.WebKitBlobBuilder ||
+                            window.MozBlobBuilder ||
+                            window.MSBlobBuilder
+                        );
+                        if (e.name === 'TypeError' && BlobBuilder) {
+                            var bb = new BlobBuilder();
+                            bb.append(request.response);
+                            blb = bb.getBlob();
+                        }
+                    }
+                    // If the blob is empty for some reason consider the image load a failure.
+                    if (blb.size === 0) {
+                        context.finish(false, "Empty image response.");
+                    } else {
+                        // Create a URL for the blob data and make it the source of the image object.
+                        // This will still trigger Image.onload to indicate a successful tile load.
+                        context.image.src = (window.URL || window.webkitURL).createObjectURL(blb);
+                    }
+                },
+                error: function(request) {
+                    context.finish(false, "Image load aborted - XHR error");
+                }
+            });
+        } else {
+            if (context.crossOriginPolicy !== false) {
+                context.image.crossOrigin = context.crossOriginPolicy;
+            }
+
+            context.image.src = context.src;
+        }
+    },
+
+    /**
+     * @param {object} context job, the same object as with downloadTileStart(..)
+     * @param successful true if successful
+     * @return {null|*} null to indicate missing data or data object
```

Comment:
Before explaining when to return `null` this comment should explain what the normal expected return value would be (image data).